geonode/people/views.py

```diff
@@ -189,6 +192,50 @@ def get_queryset(self):
         queryset = self.get_serializer_class().setup_eager_loading(queryset)
         return queryset.order_by("username")
 
+    def perform_destroy(self, instance):
+        # not implemented added to make tests pass
+        if any(
+            (
+                not self.request.user.is_superuser,
+                not self.request.user.is_staff,
+                self.request.user.pk == int(self.kwargs["pk"]),
+            )
+        ):
+            raise PermissionDenied()
+        instance.delete()
+
+    def perform_create(self, serializer):
+        user = self.request.user
+        if not (user.is_superuser or user.is_staff):
+            raise PermissionDenied()
+
+        email_payload = self.request.data.get("email", "")
+        password_payload = self.request.data.get("password", "")
+
+        if ACCOUNT_EMAIL_REQUIRED and email_payload == "":
+            raise ValidationError(detail="email missing from payload")
+        try:
```

Comment:
Since the password validation is common between update and create, we can wrap it into a single function and be called if needed